File: solutions/problem-9/solution.py
import math


def pythagorean_triplet(num):
    for a in range(1, num):
        for b in range(a, num):
            for c in range(b, num):
                if math.pow(a, 2) + math.pow(b, 2) == math.pow(c, 2) and a < b < c and a+b+c==num:
                    return a, b, c
    return None


a, b, c = pythagorean_triplet(1000)
print("Pythagorean triplets: a={}, b={}, c={}. Product: {}".format(a, b, c, a*b*c))

File: solutions/problem-9/solution.c
#include <stdio.h>
#include <math.h>
int main()
{
	int i, j, k, tempi, tempj, tempk;
	for (i=1; i<1000; i++)
	{
		for (j=1; j<1000; j++)
		{
			for (k=1; k<1000; k++)
			{
				if ((pow(i,2) + pow(j,2)) == (pow(k,2)))
				{
				       if (i+j+k == 1000)
				       {
					       tempi=i;
					       tempj=j;
					       tempk=k;
					       goto label;
				       }
				}
			}
		}
	}
	label:
		printf("Special Pythagorean Triplet = %d %d %d", i, j, k);
	return 0;
}	
